src/FSharpLint.Core/Rules/Hints/HintMatcher.fs

```diff
@@ -596,36 +597,36 @@ let private hintError typeChecks hint (args:AstNodeRuleParams) range matchedVari
         let error = System.String.Format(errorFormatString, matched, message)
         { Range = range; Message = error; SuggestedFix = None; TypeChecks = typeChecks }
 
-let private getMethodParameters (checkFile:FSharpCheckFileResults) (methodIdent:LongIdentWithDots) =
-    async {
-        let! symbol =
-            checkFile.GetSymbolUseAtLocation(
-                methodIdent.Range.StartLine,
-                methodIdent.Range.EndColumn,
-                "",
-                methodIdent.Lid |> List.map (fun x -> x.idText))
+let private getMethodParameters (checkFile:FSharpCheckFileResults) (methodIdent:LongIdentWithDots) = async {
+    let! symbol =
+        checkFile.GetSymbolUseAtLocation(
+            methodIdent.Range.StartLine,
+            methodIdent.Range.EndColumn,
+            "",
+            methodIdent.Lid |> List.map (fun x -> x.idText))
 
-        return
-            match symbol with
-            | Some(symbol) when (symbol.Symbol :? FSharpMemberOrFunctionOrValue) ->
-                let symbol = symbol.Symbol :?> FSharpMemberOrFunctionOrValue
+    match symbol with
+    | Some(symbol) when (symbol.Symbol :? FSharpMemberOrFunctionOrValue) ->
+        let symbol = symbol.Symbol :?> FSharpMemberOrFunctionOrValue
 
-                if symbol.IsMember then symbol.CurriedParameterGroups |> Seq.tryHead
-                else None
-            | _ -> None }
+        if symbol.IsMember then return symbol.CurriedParameterGroups |> Seq.tryHead
+        else return None
+    | _ -> return None
+}
 
 /// Check a lambda function can be replaced with a function,
 /// it will not be if the lambda is automatically getting
 /// converted to a delegate type e.g. Func<T>.
 let private canReplaceLambdaWithFunction checkFile methodIdent index = async {
     let! parameters = getMethodParameters checkFile methodIdent
 
-    return
-        match parameters with
-        | Some(parameters) when index < Seq.length parameters ->
-            let parameter = parameters.[index]
-            not (parameter.Type.HasTypeDefinition && parameter.Type.TypeDefinition.IsDelegate)
-        | _ -> true }
+
+    match parameters with
+    | Some(parameters) when index < Seq.length parameters ->
+        let parameter = parameters.[index]
+        return not (parameter.Type.HasTypeDefinition && parameter.Type.TypeDefinition.IsDelegate)
```

Comment:
This is the line breaking the tests, in the failing tests the method's argument don't seem to have the correct type info, taking a closer a look

Reply:
We're getting `(5,25)-(5,31) typecheck error The type 'string' is not defined.` in our `FSharpCheckFileResults`. Guessing the root of it lies with `GetProjectOptionsFromScript`

Reply:
`GetProjectOptionsFromScript` 😱

Reply:
good grief. this function has been a pain every since .net core became a thing. @duckmatt it's so hilariously wrong that for FSAC we've completely replaced the reference probing.  if at all possible can you dump the project options that are discovered for this test run?

Reply:
The output doesn't look too bad:

```
{ ProjectFileName = "test.fsx.fsproj"
  ProjectId = None
  SourceFiles = [|"test.fsx"|]
  OtherOptions =
                [|"--noframework"; "--warn:3";
                  "-r:C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Numerics.dll";
                  "-r:C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.dll";
                  "-r:C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\netstandard.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\Microsoft.CSharp.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\Microsoft.VisualBasic.Core.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\Microsoft.VisualBasic.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\Microsoft.Win32.Primitives.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\mscorlib.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\netstandard.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.AppContext.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Buffers.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.Concurrent.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.Immutable.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.NonGeneric.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.Specialized.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.Annotations.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.DataAnnotations.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.EventBasedAsync.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.Primitives.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.TypeConverter.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Configuration.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Console.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Core.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Data.Common.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Data.DataSetExtensions.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Data.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Contracts.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Debug.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.DiagnosticSource.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.FileVersionInfo.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Process.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.StackTrace.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.TextWriterTraceListener.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Tools.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.TraceSource.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Tracing.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Drawing.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Drawing.Primitives.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Dynamic.Runtime.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Formats.Asn1.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Globalization.Calendars.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Globalization.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Globalization.Extensions.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Compression.Brotli.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Compression.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Compression.FileSystem.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Compression.ZipFile.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.FileSystem.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.FileSystem.DriveInfo.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.FileSystem.Primitives.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.FileSystem.Watcher.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.IsolatedStorage.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.MemoryMappedFiles.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Pipes.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.UnmanagedMemoryStream.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Linq.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Linq.Expressions.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Linq.Parallel.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Linq.Queryable.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Memory.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Http.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Http.Json.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.HttpListener.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Mail.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.NameResolution.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.NetworkInformation.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Ping.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Primitives.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Requests.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Security.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.ServicePoint.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Sockets.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebClient.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebHeaderCollection.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebProxy.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebSockets.Client.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebSockets.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Numerics.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Numerics.Vectors.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ObjectModel.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.DispatchProxy.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Emit.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Emit.ILGeneration.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Emit.Lightweight.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Extensions.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Metadata.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Primitives.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.TypeExtensions.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Resources.Reader.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Resources.ResourceManager.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Resources.Writer.dll";
                  "-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.CompilerServices.Unsafe.dll";
                  ...|]
  ReferencedProjects = [||]
  IsIncompleteTypeCheckEnvironment = false
  UseScriptResolutionRules = true
  LoadTime = 31/12/9999 23:59:59
  UnresolvedReferences = Some UnresolvedReferencesSet []
  OriginalLoadReferences = []
  ExtraProjectInfo = None
  Stamp = None }
```

These are the errors from the type check:

```
[|startup (1,1)-(1,1) parameter error The type 'WebClient' is required here and is unavailable. You must add a reference to assembly 'System.Net.WebClient, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.;
  startup (1,1)-(1,1) typecheck error The type 'WebClient' is required here and is unavailable. You must add a reference to assembly 'System.Net.WebClient, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.;
  test.fsx (5,39)-(5,45) typecheck error The type 'string' is not defined.;
  test.fsx (5,39)-(5,45) typecheck error The type 'string' is not defined.|]
```

Reply:
There is a reference to `System.Net.WebClient.dll` in the project options which does point to an existing file on my machine 🤔

Reply:
the mscorlib.dll dependency in there is worrisome, as are the multiple instances of `netstandard.dll`. Can you dump the entire `OtherOptions`?  you'll have to do your own `String.concat "\n" opts.OtherOptions` equivalent because the %A formatting stops printing after a certain length of array/

Reply:
Ah, apologies, didn't spot it had been trimmed. Updated results below:

```
--noframework
--warn:3
-r:C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Numerics.dll
-r:C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.dll
-r:C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\netstandard.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\Microsoft.CSharp.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\Microsoft.VisualBasic.Core.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\Microsoft.VisualBasic.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\Microsoft.Win32.Primitives.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\mscorlib.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\netstandard.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.AppContext.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Buffers.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.Concurrent.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.Immutable.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.NonGeneric.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Collections.Specialized.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.Annotations.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.DataAnnotations.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.EventBasedAsync.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.Primitives.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ComponentModel.TypeConverter.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Configuration.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Console.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Core.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Data.Common.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Data.DataSetExtensions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Data.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Contracts.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Debug.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.DiagnosticSource.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.FileVersionInfo.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Process.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.StackTrace.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.TextWriterTraceListener.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Tools.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.TraceSource.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Diagnostics.Tracing.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Drawing.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Drawing.Primitives.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Dynamic.Runtime.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Formats.Asn1.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Globalization.Calendars.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Globalization.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Globalization.Extensions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Compression.Brotli.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Compression.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Compression.FileSystem.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Compression.ZipFile.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.FileSystem.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.FileSystem.DriveInfo.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.FileSystem.Primitives.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.FileSystem.Watcher.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.IsolatedStorage.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.MemoryMappedFiles.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.Pipes.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.IO.UnmanagedMemoryStream.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Linq.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Linq.Expressions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Linq.Parallel.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Linq.Queryable.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Memory.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Http.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Http.Json.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.HttpListener.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Mail.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.NameResolution.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.NetworkInformation.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Ping.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Primitives.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Requests.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Security.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.ServicePoint.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.Sockets.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebClient.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebHeaderCollection.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebProxy.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebSockets.Client.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Net.WebSockets.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Numerics.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Numerics.Vectors.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ObjectModel.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.DispatchProxy.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Emit.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Emit.ILGeneration.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Emit.Lightweight.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Extensions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Metadata.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.Primitives.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Reflection.TypeExtensions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Resources.Reader.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Resources.ResourceManager.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Resources.Writer.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.CompilerServices.Unsafe.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.CompilerServices.VisualC.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Extensions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Handles.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.InteropServices.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.InteropServices.RuntimeInformation.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Intrinsics.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Loader.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Numerics.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Serialization.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Serialization.Formatters.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Serialization.Json.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Serialization.Primitives.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Runtime.Serialization.Xml.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.Claims.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.Cryptography.Algorithms.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.Cryptography.Csp.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.Cryptography.Encoding.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.Cryptography.Primitives.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.Cryptography.X509Certificates.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.Principal.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Security.SecureString.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ServiceModel.Web.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ServiceProcess.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Text.Encoding.CodePages.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Text.Encoding.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Text.Encoding.Extensions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Text.Encodings.Web.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Text.Json.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Text.RegularExpressions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.Channels.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.Overlapped.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.Tasks.Dataflow.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.Tasks.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.Tasks.Extensions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.Tasks.Parallel.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.Thread.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.ThreadPool.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Threading.Timer.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Transactions.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Transactions.Local.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.ValueTuple.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Web.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Web.HttpUtility.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Windows.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.Linq.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.ReaderWriter.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.Serialization.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.XDocument.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.XmlDocument.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.XmlSerializer.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.XPath.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\System.Xml.XPath.XDocument.dll
-r:C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.0\../../../packs/Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0\WindowsBase.dll
-r:C:\Users\p4rk0\github\FSharpLint\src\FSharpLint.Console\bin\Debug\net5.0\FSharp.Core.dll
```

Reply:
Odd how System.Windows and WindowsBase end up in that assembly list. are those netstandard references?

Another thing that's concerning me is that all three 'primary assemblies' are in this resultset. Normally the 'primary assembly' is determined from your `targetprofile`, and you should only have one of the three in any given set of assembly references. The mapping is as follows:

| target profile | primary assembly |
| ---------| --------------|
| mscorlib | mscorlib.dll |
| netstandard | netstandard.dll |
| netcore | System.Runtime.dll |

Given that, this _may_ be an FCS bug but I'm not sure.

Reply:
I've found something, this line:

https://github.com/fsharp/FSharp.Compiler.Service/blob/122520fa62edec7be5d00854989b282bf3ce7315/src/fsharp/DotNetFrameworkDependencies.fs#L351

Is grabbing all files under my `C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0\ref\net5.0` directory, this is including mscorlib.dll, netstandard.dll, and System.Runtime.dll.

I've set `useSdkRefs` to false and it has resolved the issue, although it doesn't feel like a "true" fix 

WindowsBase doesn't seem to be part of netstandard, but has ended up in that sdk folder seemingly as part of wpf for core